### PR TITLE
feat(assistant_demo): event socket example

### PR DESCRIPTION
**Pull request by unique-fabian**

### CODE REVIEW

**Review by mfadler — COMMENTED** [review]
Few smaller comments.

File: assistant_demo/assistant_demo/event_socket.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+logger = logging.getLogger(__name__)
+
+# Load environment variables
+load_dotenv("./.env", override=True)
+
+unique_sdk.api_key = os.environ.get("API_KEY")
+unique_sdk.app_id = os.environ.get("APP_ID")
+
+
+def _get_sse_client():
```
[the remark this answers is not in the document]
Comment (unique-fabian) [reply assistant_demo/assistant_demo/event_socket.py]: Removed the underscore

File: assistant_demo/assistant_demo/event_socket.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    return SSEClient(
+        url=f'{os.getenv("BASE_URL")}/public/event-socket/events/stream?subscriptions={SUBSCRIPTIONS}',
```
[the remark this answers is not in the document]
Comment (unique-fabian) [reply assistant_demo/assistant_demo/event_socket.py]: added a env var check function

File: assistant_demo/assistant_demo/event_socket.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            "x-app-id": os.getenv("APP_ID"),
+            "x-company-id": os.getenv("COMPANY_ID"),
+        },
+    )
+
+
+def _process_event(event_data):
+    try:
+        event = json.loads(event_data)
+        print(event)
```
[the remark this answers is not in the document]
Comment (unique-fabian) [reply assistant_demo/assistant_demo/event_socket.py]: done

File: assistant_demo/assistant_demo/event_socket.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    except json.JSONDecodeError as e:
+        logger.error(f"JSON decoding error: {e}")
+        return
+
+    if (
+        event
+        and event["event"] in SUBSCRIPTIONS.split(",")
+        and event["payload"]["assistantId"] == os.getenv("ASSISTANT_ID")
```
[the remark this answers is not in the document]
Comment (unique-fabian) [reply assistant_demo/assistant_demo/event_socket.py]: done

File: assistant_demo/assistant_demo/event_socket.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+unique_sdk.app_id = os.environ.get("APP_ID")
+
+
+def _get_sse_client():
+    return SSEClient(
+        url=f'{os.getenv("BASE_URL")}/public/event-socket/events/stream?subscriptions={SUBSCRIPTIONS}',
+        headers={
+            "Authorization": f'Bearer {os.getenv("API_KEY")}',
+            "x-app-id": os.getenv("APP_ID"),
+            "x-company-id": os.getenv("COMPANY_ID"),
```
[the remark this answers is not in the document]
Comment (unique-fabian) [reply assistant_demo/assistant_demo/event_socket.py]: done

File: assistant_demo/assistant_demo/event_socket.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+        and event["payload"]["assistantId"] == os.getenv("ASSISTANT_ID")
+    ):
+        logger.info(f"Event subscription received: {SUBSCRIPTIONS}")
+
+        # Send a message back to the user
+        unique_sdk.Message.create(
+            user_id=event["userId"],
+            company_id=event["companyId"],
+            chatId=event["payload"]["chatId"],
+            assistantId=os.getenv("ASSISTANT_ID"),
```
[the remark this answers is not in the document]
Comment (unique-fabian) [reply assistant_demo/assistant_demo/event_socket.py]: done

**Review by mfadler — APPROVED** [review]
Looks good now. Thank you for integrating my suggested changes.

**Pull request merged**